### PR TITLE
fix: allow for absolute image URLs in the image normalizer

### LINKED ISSUE #5559: Issue with complete media URL
if from Backend we receive full product image URL then in `ProductImageNormalizer` it is concatenated with baseURL. Which is wrong.
for example if image have following url 

`https://externalImageServer/images/product/001.jpg`
and base Url ist `localhost:9002` then the final image url is produces as 

`https://localhost:9002https://externalImageServer/images/product/001.jpg`

Problem is here in class --> ProductImageNormalizer at linke 57
```
image.url =
          (this.config.backend.media.baseUrl ||
            this.config.backend.occ.baseUrl ||
            '') + image.url;
```
## Environment Details

- Spartacus:  1.3.*

## Offer to Fix
YES

### PR DESCRIPTION
**Pull request by mrSingh007**
closes #5559

### CODE REVIEW

File: projects/core/src/occ/adapters/product/converters/product-image-normalizer.ts [hunk 1]
```diff
@@ -55,14 +55,14 @@ export class ProductImageNormalizer implements Converter<Occ.Product, Product> {
...
-            '') + image.url;
-
+        targetImage.url = this.processTargetImageUrl(targetImage.url);
         imageContainer[image.format] = targetImage;
       }
     }
     return images;
   }
+
+  private processTargetImageUrl(url: string){
```
Comment (tobi-or-not-tobi) [inline projects/core/src/occ/adapters/product/converters/product-image-normalizer.ts]: I'd rename this to `normalizeImageUrl()`.

Comment (tobi-or-not-tobi) [inline projects/core/src/occ/adapters/product/converters/product-image-normalizer.ts]: The inline documentation (line 51 - 57) can now move to the method documentation.

File: projects/core/src/occ/adapters/product/converters/product-image-normalizer.ts [hunk 2]
```diff
@@ -55,14 +55,14 @@ export class ProductImageNormalizer implements Converter<Occ.Product, Product> {
...
+    return  url.startsWith('http') ? url : (this.config.backend.media.baseUrl || this.config.backend.occ.baseUrl || '') + url;
```
Comment (tobi-or-not-tobi) [inline projects/core/src/occ/adapters/product/converters/product-image-normalizer.ts]: I'd prepare for more cases here:
- uppercase / lowercase/ mixed case
- the `//` syntax (allows for cross http / https)

We could even consider other protocols or the `data:image` for data URLs, you could evaluate an array of prefixes and test against it (_contains_) Given that we're concerned with the OCC backend, those are really edges cases though. Up to you!

File: projects/core/src/occ/adapters/product/converters/product-image-normalizer.spec.ts [hunk 3]
```diff
@@ -37,6 +37,12 @@ describe('ProductImageNormalizer', () => {
         imageType: ImageType.PRIMARY,
         url: '/test2',
       },
+      {
+        altText: 'Test alt text',
+        format: 'zoom',
+        imageType: ImageType.PRIMARY,
+        url: 'https://hybris.com/demo.jpg',
```
Comment (tobi-or-not-tobi) [inline projects/core/src/occ/adapters/product/converters/product-image-normalizer.spec.ts]: I'd at least also include a variation for `http` and  `//`, and the uppercase/lowercase variants. Even if you're code expects to start with http, a future refactor might change the code and the http variation might fail in production.

**Comment by giancorderoortiz** [comment]
Closing due to many months of inactivity.
If still relevant, please update PR with latest develop branch, and re-open with meaningful comments so that it can be evaluated by development team.

**Pull request closed without merging**